Teams - main branch (Make PRs against this branch!) (#140)
mulesoft/docs-access-management: modules/ROOT/pages/users.adoc

include::_attributes.adoc[]
endif::[]

You must have the Organization Administrator permission to manage users within an Anypoint Platform organization.
To manage user permissions for your API version, you must have the API Version Owner permission.

As an administrator of your root organization, you can enable, disable, or delete users.

> By ticking the checkbox next to a user, additional options related to that user are displayed:

Reviewer: ```suggestion
By selecting the checkbox next to a user, additional options related to that user are displayed:
```

> * *Enable*: Enables the user in the selected Organization / Business Group.

Author: ```suggestion
* *Enable*: Enables the user in the selected organization or business group.
```

> * *Disable*: Disables the user in the selected Organization / Business Group. They will no longer be able to log in.

Reviewer: What does  /  indicate? And? Or? best to spell if not a label or name....
```suggestion
* *Disable*: Disables the user in the selected organization or business group. They are no longer able to log in.
```
...but do we really need that in light of the sentence directly before?
Does it also disable any current sessions?

When running Anypoint Platform Private Cloud Edition, you need to configure user management methods such as SSO through a third-party IdP or LDAP.
--

> As an organization administrator of a root organization or of one of its business groups, you can invite new users and manage existing users for your organization.

Reviewer: Is "an organization administrator" = having "the Organization Administrator permission"?
We seem to make this distinction throughout, about having the permission vs. having that title or job....

To invite new users to your organization:

include::partial$include-nav-users.adoc[]

. Click *Invite user*.

> . Enter the email addresses of who you want to invite in your organization as a comma-separated list.

Reviewer: ```suggestion
. Enter the email addresses of the users that you want to invite to your organization as a comma-separated list.
```

> . Optionally, select a xref:teams.adoc[team] or xref:roles.adoc[role (deprecated)] to which to assign these users.

Reviewer: ```suggestion
. Optionally, select a xref:teams.adoc[team] or xref:roles.adoc[role (deprecated)] for assign these users.
```
Should we explain "deprecated"? Does the UI still show this but it is understood to equal "permissions" instead?

Author: Roles deprecation is explained in the Roles topic. We are transitioning to no longer support roles and request that they opt in to Teams. From there, Roles still exist but are deemphaized. It changes the whole UI when you opt in to teams vs continuing to use roles. But in this case, both of these appear in this screen. Roles has (deprecated) next to it in this screen too fwiw although I didn't call them out as labels.

. Click *Send Invitation*.

> The users who you invite receive email invitations to sign up to your organization. Invited users must use the link they receive in the invitation email to join your organization. When they click through the link they are presented with a sign-up form that already has the *Company* field completed, matching your organization.

Reviewer: ```suggestion
The users you specify receive email invitations to join your organization. Invited users must use the link they receive in the invitation to join your organization. When they click the link, they are presented with a sign-up form that already has the *Company* field completed with the name of your organization.
```

Reviewer: I wonder if I could have this PR in reserve for times that I don't have a lot of editing to do, to clean it up for you? Ask Meeta; dunno' if this is a priority doc....

Note that the email link expires in one week.

> Invited users have access to the same set of resources as you (although they may have different permissions that can restrict what they can view or do).

Reviewer: ```suggestion
Invited users have access to the same set of resources as you, although they likely have different permissions.
```
This I don't understand: the difference between access and permissions. If you have access, don't you have permission for that access?

Author: This is confusing, but for example, Exchange Viewer and Exchange Contributor: both let you access and subsequently view a resource, but the former won't let you mess with them. So you can access both, but you don't have permission to play with the resource. So no, it is not the same.

== Resending or Canceling a Sent Invitation

Click *Pending Invites* to view all invitations that have not yet been accepted. Select the ones you need to manage, and click *Re-send Invite* or *Cancel Invite*.

[[grant-user-permissions]]
== Grant Permissions and Roles to Users

> Organization administrators can manage user access using xref:teams.adoc[teams], xref:roles.adoc[roles] (deprecated), or by granting permissions to users individually.

Reviewer: To remove modal verb...:
```suggestion
Organization administrators manage user access using xref:teams.adoc[teams] or xref:roles.adoc[roles] (deprecated), or by granting permissions to users individually.
```
Same questions: having permissions vs. having that title/job; explanation of "deprecated"

> In the *Users* page, click a user name to access more information about that user, grant roles and permissions, see inherited permissions, or reset their password.

Reviewer: Is it okay to use "roles and permissions" here when we are deprecating roles in favor of permissions?

Author: We are deprecating roles in favor of teams. Individual permissions have always been supported and somewhat discouraged but necessary to call out. They can still grant and revoke roles and permissions in this UI. But they can't mess with Teams here, afaik.

Roles and permissions are grouped under organizations (and also, optionally, under business groups). This means that you can only assign roles and permissions that are related to resources that exist in the organization and/or business group that you are selecting.
--

> If necessary, you can also remove user permissions, but note that if users have been granted their permissions through an assigned role, you can remove role-based permissions using only the xref:roles.adoc[Roles] functionality. You cannot view or remove role-based permissions in the *Users* page.

Reviewer: ```suggestion
If necessary, you can also remove user permissions, but note that if those permissions are granted through an assigned role, you can remove them only by using the xref:roles.adoc[Roles] functionality. You cannot view or remove role-based permissions in the *Users* page.
```
But...CAN you remove them using the Roles functionality, if that functionality is deprecated?

Author: For now, yes.

> Because the Roles feature is deprecated, it is best to xref:teams.adoc#opt-in-to-teams[opt in to the Teams feature] to manage permissions in a hierarchy.

Reviewer: Maybe the entire topic should begin with this: why discuss the ins and outs of deprecated features, if the audience can't or won't be using those features?

Author: The most probable use-case is that someone will be migrating from roles to teams over the next few months incrementally. As far as the IA for this repo goes, it does not make sense to disambiguate the roles deprecation in this topic.

To grant permissions to an individual user:

include::partial$include-nav-steps-org-admin.adoc[]

include::partial$include-nav-users.adoc[]

. Select the user whose permissions you want to modify.
. Select the product that contains the permission you want to add. For more information about the permissions that are available across Anypoint Platform, see xref:permissions-by-product.adoc[Permissions by Product].
* If you want to grant the user permissions that apply across the current business group:

> .. Select the permissions from the top *Permission(s)* drop-down.

Reviewer: What does "the top" mean? At the top of the page/window/section...? Also, let's GLOBALLY replace "drop-down" with menu, assuming that our audiences know how to identify and use common UI controls.

> .. Select the + button.

Reviewer: ```suggestion
.. Click *+*.
```

> * If you want to apply a permission only across a specific environment:

Reviewer: ```suggestion
* If you want to apply a permission across a specific environment only:
```

> .. Select the environments using the *Environments* drop-down.

Reviewer: ```suggestion
.. Select the environment using the *Environments* menu.
```

> .. Select the permissions from the *Permission(s)* drop-down that is located next to the *Environment(s)* drop-down.

Reviewer: ```suggestion
.. Select the permissions from the *Permission(s)* menu.
```

> .. Select the + button.

Reviewer: ```suggestion
.. Click *+*.
```

> . Select the user whose permissions you want to modify.
> . Select the product that contains the permission you want to add. For more information about the permissions that are available across Anypoint Platform, see xref:permissions-by-product.adoc[Permissions by Product].
> * If you want to grant the user permissions that apply across the current business group:
> .. Select the permissions from the top *Permission(s)* drop-down.
> .. Select the + button.
> * If you want to apply a permission only across a specific environment:
> .. Select the environments using the *Environments* drop-down.
> .. Select the permissions from the *Permission(s)* drop-down that is located next to the *Environment(s)* drop-down.
> .. Select the + button.

Author: ```suggestion
. In the *Permissions* section, click *Add permissions*.
. Next to each permission you want to grant the user, select the checkbox.
. Click *Next*.
. Select the business groups in which you want the selected permissions to apply.
. Click *Next*.
. Select the environments in which you want the selected permissions to apply.
. Review the permissions, business groups, and environments, and then click *Add permissions*.

The user now has the permissions you selected in the business groups and environments you specified.
```
